src/mlpack/methods/kde/kde_stat.hpp

```diff
@@ -48,6 +49,128 @@ class KDEStat
     {
       validCentroid = false;
     }
+
+    // Build orthogonal base if possible.
+    if (tree::TreeTraits<TreeType>::BinaryTree)
```

Comment:
I see that this is specific to binary trees, and the reason seems to be that the subspace merging is done for two subspaces at a time only.  Maybe you have already thought of doing this (the PR is a WIP after all) but it seems to me that you could handle a node with three children by merging the subspaces of child 0 and 1, and then subsequently merging that resulting subspace with the subspace of child 2.

In any case, probably best to get it working just with the kd-tree first. :)